File: MetodosOrdenamiento/burbuja/burbuja.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

function ordenamientoBurbuja(arr) {
    let n = arr.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
            }
        }
    }
    return arr;
}

readline.question('Ingrese el tamaño del arreglo: ', (tamaño) => {
    // Crear arreglo con valores aleatorios
    let arreglo = [];
    for (let i = 0; i < tamaño; i++) {
        arreglo.push(Math.floor(Math.random() * 100) + 1);
    }

    console.log("Arreglo original:", arreglo);

    // Ordenar el arreglo (usando copia)
    let arregloOrdenado = ordenamientoBurbuja([...arreglo]);

    console.log("Arreglo ordenado:", arregloOrdenado);

    readline.close();
});

File: MetodosOrdenamiento/burbuja/burbuja.py
import random

def ordenamiento_burbuja(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

tamaño = int(input("Ingrese el tamaño del arreglo: "))

# Crear arreglo con valores aleatorios
arreglo = [random.randint(1, 100) for _ in range(tamaño)]

print(f"Arreglo original: {arreglo}")

# Ordenar el arreglo (usando copia para no modificar el original)
arreglo_ordenado = ordenamiento_burbuja(arreglo.copy())

print(f"Arreglo ordenado: {arreglo_ordenado}")